utils/entrypoint.sh

```diff
@@ -167,6 +176,25 @@ zm_db_exists() {
     fi
 }
 
+# The secret sauce to determine wether to use mysql_install_db
+# or mysqld --initialize seems to be wether mysql_install_db is a shell
+# script or a binary executable
+use_mysql_install_db () {
+    MYSQL_INSTALL_DB=$(type mysql_install_db)
+    local result="$?"
+
+    if [ "$result" -eq "0" ] && [ -n $MYSQL_INSTALL_DB  ]; then
```

Comment:
hm.
```
 line 186: [: too many arguments
```

try to understand the issue.

Reply:
line 187 in log that's why me added the first line at entrypoint.sh `set -x`
```
zoneminder | ++ mysql_datadir_exists
zoneminder | ++ '[' -d /var/lib/mysql/mysql ']'
zoneminder | ++ echo 0
zoneminder | + '[' 0 -eq 0 ']'
zoneminder | + echo ' * First run of MYSQL, initializing DB.'
zoneminder |  * First run of MYSQL, initializing DB.
zoneminder | ++ use_mysql_install_db
zoneminder | +++ type mysql_install_db
zoneminder | ++ MYSQL_INSTALL_DB='mysql_install_db is /usr/bin/mysql_install_db'
zoneminder | ++ local result=0
zoneminder | ++ '[' 0 -eq 0 ']'
zoneminder | ++ '[' -n mysql_install_db is /usr/bin/mysql_install_db ']'
zoneminder | /usr/local/bin/entrypoint.sh: line 187: [: too many arguments
zoneminder | ++ echo 0
zoneminder | + '[' 0 -eq 0 ']'
zoneminder | + --user=mysql --datadir=/var/lib/mysql/
```

Reply:
i think you should use `-n "$MYSQL_INSTALL_DB"`

Reply:
Fixed